test/unit/RedStoneOracle.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.17;
+import {RedstoneConsumerNumericBase} from "@redstone-finance/evm-connector/contracts/core/RedstoneConsumerNumericBase.sol";
+
+import {CommonTest} from "../Common.sol";
+
+import {BaseController} from "../../src/controllers/BaseController.sol";
+import {ChainlinkSourceAdapter} from "../../src/adapters/source-adapters/ChainlinkSourceAdapter.sol";
+import {DecimalLib} from "../../src/adapters/lib/DecimalLib.sol";
+import {IAggregatorV3Source} from "../../src/interfaces/chainlink/IAggregatorV3Source.sol";
+import {RedstonePriceFeedWithRounds} from "../../src/oracles/RedstonePriceFeedWithRounds.sol";
+
+import {IAggregatorV3Source} from "../../src/interfaces/chainlink/IAggregatorV3Source.sol";
+
+import {TestedSourceAdapter} from "../fork/adapters/ChainlinkSourceAdapter.sol";
+
+contract MockRedstonePayload is CommonTest {
+    function getRedstonePayload(
+        string memory priceFeed
+    ) public returns (bytes memory) {
+        string[] memory args = new string[](4);
+        args[0] = "node";
+        args[1] = "--no-warnings";
+        args[2] = "./redstone/getRedstonePayload.js";
+        args[3] = priceFeed;
+
+        return vm.ffi(args);
+    }
+}
+
+contract RedstoneOracleAdapterTest is CommonTest, MockRedstonePayload {
+    RedstonePriceFeedWithRounds redstoneOracle;
+    TestedSourceAdapter sourceAdapter;
+
+    function setUp() public {
+        redstoneOracle = new RedstonePriceFeedWithRounds(bytes32("BTC"));
+        sourceAdapter = new TestedSourceAdapter(
+            IAggregatorV3Source(address(redstoneOracle))
+        );
+    }
+
+    function pushPrice() internal returns (uint256) {
+        bytes memory data = getRedstonePayload("BTC");
+
+        (
+            bytes memory redstonePayload,
+            uint256 timestampMilliseconds,
+            uint256 updatePrice
+        ) = abi.decode(data, (bytes, uint256, uint256));
+
+        bytes memory encodedFunction = abi.encodeWithSignature(
+            "updateDataFeedsValues(uint256)",
+            timestampMilliseconds
+        );
+        bytes memory encodedFunctionWithRedstonePayload = abi.encodePacked(
+            encodedFunction,
+            redstonePayload
+        );
+
+        address(redstoneOracle).call(encodedFunctionWithRedstonePayload);
+
+        return updatePrice;
+    }
+
+    function testPushPrice() public {
+        vm.warp(1234);
```

Comment:
The timestamp associated to the price pushed is the `block.timestamp` when the price is pushed.

Reply:
yup, same as with chainlink.